<!DOCTYPE html>

<html lang="en">

<!-- HEAD -->

   <head>
      <meta charset="UTF-8">
      <meta name="viewport" content="width=device-width, initial-scale=1.0">
      <title> TrimTech </title>
      <link rel="icon" type="image/x-icon" href="static/favicon.ico">
      <link rel="stylesheet" href="static/style.css"/>
   </head>

<body>
   <!-- HEADER -->

   <header id="navigation_bar">
      <a href="homepage.html">
         <img id="logo" src="static/logo_trimtech.png" alt="website logo">
      </a>

      <nav id="text_navbar">
         <a href="about.html"> ABOUT </a>
         <a href="uploadpage.html"> UPLOAD </a>
         <a href="fastqc.html"> FASTQC </a>
         <a href="trimmomatic.html"> TRIMMOMATIC </a>
         <a href="contactpage.html"> CONTACT </a>
      </nav>

    </header>


    <!-- CONTENT -->
    <h3 class="head_text_fastqc_page"> FASTQC </h3>
    <div class="text_fastqc_page">
         <h1>What is FastQC</h1>
         <p>FastQC is a visualisation tool for FastQ files. This tool visualizes some of the provided data 
         so you can do some quality control checks. The aim of this tool is to potentialy identify problems that could have
         occord with the sequencer itself or the sample that was used in the test.
         </p>
 
         <h2>Plot Information</h2>
 
         <h3>Basic statistics</h3>
         <p>This plot contains some basic information about the file.</p>
         
         <h3>Per base sequence quality</h3>
         <p>This boxplot contains the quality per base </br>
         Frequent problems</br>
         its commmon for reads to degrade in quality over the duration in long runs.
         There are a couple of diffrent types of encoding that can be applied on the file, this can lead to false reports on
         the this plot.
         </p>
         
         <h3>Per sequence quality score</h3>
         <p>This graph shows the overal quality distribution of the reads.</br>
         if the overal quality is low there might be systematic problem with the flow cell or a specific part of the run.
         As per babraham bioinformatics standerds if the overal quality is below 27 the read is flagd for warning and if its below 20
         its flagd for failure
         </p>
         
         <h3>Per base sequence content</h3>
         <p>In this graph you can see the proportions of the different nucleotide bases</br>
             In most cases there should be no bias between the complementing bases threwout the run, its worth mentioning that certain
             sample's can produces biases in the beginning of a run based on the way they where prepared.</br>
             if the AT or the CG compensition is more then 10% appart it wil be flagd for warning and if its greater then 20% it will be flagd
             for failure.</br>
             Primers,adapters and some samples based on their preparation can create baises.
         </p>
         
         <h3>Per sequance GC content</h3>
         <p>In this plot you can see the CG content across the sequence</br>
         The peak of this graph should match the overal CG content of the genome you are testing. If this is not the case
         or there are multiple peaks the sample can be contaminated. The theoretical distribution is calculated based on the provided
         if this is weirdly shaped it can also indicate that the sample was contaminated.
         this plot wil be flag for warning the total amount of diviations from the normal distribution exceeds 15% and if it exceeds 30% its
         flagd for failure.
         </p>
         
         <h3>Per base N content</h3>
         <p>Sequencers cant always decides with confidinces wich of the bases it is and with substitutes
         it for N. Its normal for smal amounts of N to apear becouse of quality loss near the end of the sequence. The 2 most commmon
         reasons for a large portion of N content is a bad quality or a biased sample in wich case the sequencer can get confused and make 
         bad calls.
         </p>
         
         <h3>Sequence lengt distribution</h3>
         <p>This plot shows the distribution of the lengt of the reads.</br>
             Warnings will apear when the sequance is not of the same lengt and it will be faild if the lenght is 0.
             Warnings in this graph can sometimes be ignored if you expect diffrent lengts in sequence, this could be becouse of the type of
             sequencing you have done or data trimming wich can result into a diffrence in lengt.
         </p>
         
         <h3>Sequence duplication graph</h3>
         <p> This graph shows the amount of duplicate sequences. if there are more then 20% duplicate sequences in total this graph will be 
             flagd for a warning and for failure if its over 50%. There are many reasons why duplications can arise the most common examples
             are: enricht samples or by chance the same sequence was selected.</br> 
             for more information on this graph and how to interpret unexpected results please visit <a href="https://www.bioinformatics.babraham.ac.uk/projects/fastqc/Help/3%20Analysis%20Modules/8%20Duplicate%20Sequences.html">Duplication Explanation</a>
             as it contains an extensive explanation on this graph.
         </p>
         
         <h3>Overrepresented sequences</h3>
         <p>This graph shows overrepresentation of the sequance. there are 3 main reasons why a sequance can be overrepresented, one of them 
             is that the sample is contaminated and the second one is lack of diversity. And the last reason is that the sequance can be biologically
             significant 
         </br>
             this module will show a warning any part of the sequence represents more then 0.1% of and shows a failure at more then 1%
         </p>
         
         <h3>Adapter content</h3>
         <p>The adapters of your sequence sample can influence overrepresentation in your data, this graph will show specific the specific
             adapter sequences 
         </br>
 
         </p>
 
         <h2>Source</h2>
         <p>If you require more detailt information please visit the official FastQC website</p>
         <a href="https://www.bioinformatics.babraham.ac.uk/projects/fastqc/Help/">Babraham Bioinformatics</a>
    </div>


       <!-- FOOTER -->

   <footer>
    <div class="footer">
       <div class="text_footer">
          <p> &copy;TrimTech ALL RIGHTS RESERVED</p>
       </div>
    </div>
 </footer>

</body>
</html>