autors: Ivar Lottman, Storm Steller, Kasthury Inparajah, Mirte Draaijer 
version: 0.1
date:

Name and usage
Toolbox trimsequance
This website is used for quality control and trimming fastQ files before the mapping and coverage fase of the data.
We created this website becouse not all of the data produced by sequancing is a 100% acurate and when you want to do mapping and covarige of a refrance genome you want to use acurate data. Therefor this tool was created to help in that process. 

It does this by visualising the fastq data with a tool cald fastQC and trimommatic for trimming the fastQ data based on input commands

Programmer/ICT user manual
#to do manual based on how the GUI works on the website

GUI input file
this part of the gui is for inputting files

fastQC based commands
#to do 
boxplot comman

Trimommatic based commands
#to do 
addapter
cropping
lenght

instalation # expanded manual
FastQC
Trimommatic

Systeem requirements
Working OS: 
Windows 10 or 11
linux Debian v12

Hardware requirements
No specific requirements

Instalation Manual short

Java v1.8.0_401
Follow the instal manual from the java website
https://www.java.com/nl/download/

Python v3.11.8
Follow the instal manual from the python 

https://www.python.org/downloads/

FastQC v0.12.1
https://www.bioinformatics.babraham.ac.uk/projects/fastqc/

Trimmomatic v0.39
http://www.usadellab.org/cms/?page=trimmomatic

Module discription
Commandline: Scriptname

#to do per script
Website script
Tool script
Flask # Version

Support
i.lottman@st.hanze.nl
s.steller@st.hanze.nl
k.inparajah@st.hanze.nl
mi.draaijer@st.hanze.nl

Acknowledgment
#to do