<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <title>Trimmomatic</title>
</head>
<body>
    <h1> Trimmomatic </h1>


    <h2> What is Trimmomatic? </h2>
    <p>
        With the presence of poor quality or technical sequences
        such as adapters in next-generation sequencing, data can
        result in inaccurate analyses. It is not easy to identify
        such sequences, including partial adapter sequences, while
        keeping accurate and valid sequence data intact.
    </p>

    <p>
        Trimmomatic is a tool designed to cut out those poor quality and
        technical sequences. It is a flexible, pair aware and
        efficient preprocessing tool, optimized for Illumina NGS
        data.
    </p>


    <h2> Algorithms in Trimmomatic</h2>
    <p>
        Trimmomatic has two methods to detect technical sequences
         within a read. The first method works by finding a
        close match between the read and a technical sequence
        given by the user. This mode is referred to as simple mode.
        This mode works for all technical sequences, including
        adapters and polymerase chain reaction primers.
    </p>

    <p>
        The second approach is referred to as palindrome mode. It
        is aimed at detecting a sequenced DNA fragment
        that is shorter than the read length, which results in
        adapter contamination on the end of the reads. This mode can
        only be used with paired end data but is more sensitive and
        specific than simple mode.
    </p>


    <h3> Simple mode </h3>
    <p>
        In simple mode, each read is scanned from the 5' end to the
        3' end to determine if any given adapters are present. It
        uses the standard seed and extend approach to find initial
        matches between the technical sequences and the reads. The
        seed does not have to match perfectly, and a user-given number
        of mismatches are tolerated. Based on this seed match, a local
        alignment will perform. If the score of the alignment is higher
        than the threshold given by the user, the aligned region and
        the remains after the alignment will be removed.
    </p>

    <figure>
        <img src="Trimmomatic_afbeeldingen/simplemode.jpg" alt="Simple Mode" title="Simple Mode">
        <figcaption> Figure 1: Simple mode (Bolger, Lohse, Usadel, 2014) </figcaption>
    </figure>

    <p>
        In the image above you see simple mode visualised. The proces
        begins with a partial overlap of the 3'end of the technical
        sequence with the 5'end of the read (A). Testing then proceeds
        by moving the supposed contaminant towards the 3'end of the
        read. In both the partial overlap (A) and the complete overlap
        at the 5'end (B), the entire read will be clipped. If the
        contaminant is found within the read (C), the bases from the
        5'end of the read to the beginning of the alignment are kept.
        The testing process continues until only a partial alignment on
        the 3'end of the read remains (D).
    </p>

    <p>
        Simple mode can detect any technical sequence at any location
        in the read, given that the alignment is long enough, and the
        read is sufficiently accurate. However, when only a short
        partial match is possible, the contaminant could not be
        reliably detectable.
    </p>


    <h3> Palindrome mode </h3>
    <p>
        As stated above, palindrome mode is aimed at detecting a
        sequenced DNA fragment that is shorter than the read length,
        a read through. When a read through occurs, both reads will
        consist of an equal number of valid bases, followed
        by contaminating sequence from the opposite adapters. Besides,
        the valid sequence within the two reads will be reverse
        complements. By detecting all three of these symptoms at once,
        adapter read-through can be identified with high sensitivity
        and specificity.
    </p>

    <figure>
        <img src="/Trimmomatic_afbeeldingen/palindromemode.jpg" alt="Palindrome Mode" title="Palindrome Mode">
        <figcaption> Figure 2: Palindrome mode (Bolger, Lohse, Usadel, 2014) </figcaption>
    </figure>

    <p>
        In the image above you see palindrome mode visualised. It begins
        with an overlap between the adapters and the start of the
        opposite reads (A). Because of the ligation of the adapters,
        this alignment would detect a read pair with no useful
        sequence information. The detection of this scenario would
        result into it dropping both reads. Testing then proceeds by
        moving the relative position of the reads backwards, testing
        for increasingly longer valid DNA fragments (B). This would
        result in trimming of both reads. Even when a small fragment
        of the adapter is overlapping, the overall alignment is easily
        sufficient to ensure reliable detection. (C). The process is
        complete when the overlapping region no longer reaches into
        the adapters (D).
    <p>
        The biggest advantage of palindrome mode is the longer alignment
        length, which ensures that the adapters can be reliably detected,
        even when read errors occur of where only a small number of
        adapter bases are present.
    </p>


    <h3> Alignment detection and scoring </h3>
    <p>
        The algorithm used for technical sequence alignments avoids the
        precalculated indexes often used in NGS alignments.
    </p>

    <p>
        Initial sequence comparisons are done using a 16-base fragment
        from each sequence. Those 16 bases are converted to a 64-bit
        integer, which is called a seed, using 4-bit code for each base.
        These seeds are then compared using a bitwise-XOR, which
        determines which bits differ between the two seeds. This will
        result in a 0000 code for each matching base, and a code with two
        1's for each mismatch. The 1's within the result are then counted
        using the popcount operation, and this count will be twice the
        number of differing bases for the 16-base fragments.
    </p>

    <p>
        If the seeds are within the distance given by the user, the full
        alignment scoring algorithm is used. Matching bases are scored as
        Log10<sup>4</sup>, while mismatches get penalized on their
        quality score, by Q/10, which can vary from 0 to 4. This wil result
        in a higher penalty for bases that appear to be highly accurate.
    </p>

    <p>
        Simple mode aligns each read against each technical sequence, using
        local alignment. This is applied by finding the highest scoring
        region within the alignment, and may exclude divergent regions on
        the ends.
    </p>

    <p>
        Palindrome mode aligns the forward and reverse reads, combined with
        their adapter sequences. It uses global alignment, which is the
        total alignment score of the overlapping region.
    </p>


    <h3> Quality filtering </h3>
    <p>
        Trimmomatic has two main quality filtering options. Both options
        use the Illumina quality score of each base position to determine
        where the read should be cut, which results in the 5' end portion
        maintained, while the sequence on the 3' end of the cut point is thrown
        away. This is ideal with typical Illumina data, which mostly have
        less quality at the 3' end. We will go into these options in the
        following sections.
    </p>


    <h3> Sliding Window quality filtering </h3>
    <p>
        The sliding window uses a regular approach. It scans from the 5'end
        of the read, and removes the 3' end when the average quality of a
        group of bases drops below a given threshold. This prevents a single
        weak base to cause the removal of possible high quality data, while
        ensuring that a consecutive series of low quality bases will cause
        trimming.
    </p>

    <h3> Maximum Information quality filtering </h3>
    <p>
        The maximum information option uses a combination of three factors
        to determine how much of a read should be kept. The first factor
        is the length threshold, whereby a read must be of at least a minimal
        length to be useful for the downstream application. Very short reads
        have little value, because they are too cryptic to be informative.
        Long reads on the other hand, can be mapped to few locations in the
        target sequence. If they cannot be uniquely mapped, because they come
        from a repetitive region, it is unlikely that a small number of extra
        bases will fix this. For reads between these extremes is the benefit
        of a few additional bases considerable, as these bases could make a
        cryptic read informative.
    </p>

    <p>
        A logistic curve is used to apply this scoring behavior, as it gives
        a flat score for extreme values, while giving a steep transition
        around the user given threshold point. Given a target length t,
        the supposed trimming to length l would give a length threshold
        score:
    </p>

    <figure>
        <img src="Trimmomatic_afbeeldingen/logisticcurvefactor1.jpg" alt="Logistic curve" title="Logistic curve">
        <figcaption> Figure 3: Formula logistic curve (Bolger, Lohse, Usadel, 2014) </figcaption>
    </figure>

    <p>
        The second factor copies coverage, and gives a linear score based
        on sequence length:
    </p>

    <figure>
        <img src="Trimmomatic_afbeeldingen/linearcurvefactor2.jpg" alt="Linear curve" title="Linear curve">
        <figcaption> Figure 4: Formula linear curve (Bolger, Lohse, Usadel, 2014) </figcaption>
    </figure>

    <p>
        This reflects that, given reasonably high accuracy bases, a longer
        read contains more information that is useful for most applications.
    </p>

    <p>
        The last factor copies the error rate, and uses the error
        probabilities from the read quality scores to determine the gathered
        possibility of errors over the read. To calculate this score, it
        takes the product of the probabilities that each base is correct:
    </p>

    <figure>
        <img src="Trimmomatic_afbeeldingen/formulaerror.jpg" alt="Formula error rate" title="formula error rate">
        <figcaption> Figure 5: Formula error rate (Bolger, Lohse, Usadel, 2014) </figcaption>
    </figure>

    <p>
        The correctness probabilities Pcorr of each base are calculated
        from the sequence quality scores. The error score usually begins as
        a high score at the start of the read, and depending on the quality
        of the read, typically drops fast at some point during the read.
    </p>

    <p>
        The maximum information algorithm determines the combined score
        of the three factors for each possible trimming position, and the
        best combined score determines how much of the read to trim. A
        user given strictness setting s, which can be set between 0 and 1,
        controls the balance between the coverage factor and the error
        rate factor. This results in the following general formula:
    </p>

    <figure>
        <img src="Trimmomatic_afbeeldingen/generalformula.jpg" alt="General formula" title="General formula">
        <figcaption> Figure 6: General formula (Bolger, Lohse, Usadel, 2014) </figcaption>
    </figure>

    <figure>
        <img src="Trimmomatic_afbeeldingen/graph.jpg" alt="General formula visualised" title="General formula visualised">
        <figcaption> Figure 7: General formula visualised (Bolger, Lohse, Usadel, 2014) </figcaption>
    </figure>

    <p>
        The image above shows how the three factors combine into a
        single score. The peak of the score is then used to
        determine the trimming point.
    </p>



    <h2> In- and Output </h2>
    <p>
        Trimmomatic works using FastQ files.
        In- and output files can be specified on the command line.
        Compressed input and output are supported using either
        .gzip or .bzip2 formats. Compression or decompression is
        applied automatically when the correct file extensions are
        used.
    </p>


    <h2> Trimming Options</h2>
    <p>
        When trimming, there are an amount of parameters you can set.
        those parameters include:
    </p>

    <ul>
        <li>ILLUMINACLIP: Cuts adapters and other illumina-specific sequences from the read.</li>
        <li>LEADING: Cuts bases off the start of a read, if it is below a threshold quality.</li>
        <li>TRAILING: Cuts bases of the end of a read, if it is below a threshold quality.</li>
        <li>SLIDINGSHOW: Performs a sliding window trimming, cuts if the average quality within the window is bellow a threshold. The first argument is the size of the window. The argument is the threshold quality.</li>
        <li>MINLEN: Drop the read if it is below the specified length.</li>
        <li>CROP: Cuts the read to a specified length.</li>
        <li>HEADCROP: Cuts the bases off the end of a read if below a threshold quality.</li>
        <li>TOPHRED33: Converts the quality score to Phred-33</li>
        <li>TOPHRED64: Converts the quality score to Phred-64</li>
    </ul>

    <p>
        Note: If no phred score is specified, Trimmomatic will
        default to phred-33.
    </p>

</body>
</html>