{% extends "basetemplate.html" %}

{% block content %}

   <!-- CONTENT -->
   <h1 class="head_text_fastqc"> WHAT IS FASTQC? </h1>
   <p class="text_fastqc">
      FastQC is a visualization tool for FastQ files. This tool visualizes some of the provided data 
      so you can do some quality control checks. The aim of this tool is to potentially identify problems that could have
      occurred with the sequencer itself or the sample that was used in the test.
   </p>
   <br>
 
   <h3 id="header_text_fastqc"> Plot Information:</h3>
   <h4 class="headers_text_fastqc">Basic statistics:</h4>
   <p class="text_fastqc">This plot contains some basic information about the file.</p>
   <br>

   <h4 class="headers_text_fastqc">Adapter content:</h4>
   <p class="text_fastqc"> The adapters of your sequence sample can influence overrepresentation in your data, this graph will show specific the specific adapter sequences. </p>
   <br>

   <h4 class="headers_text_fastqc"> Sequence duplication:</h4>
   <p class="text_fastqc">
      This graph shows the number of duplicate sequences. if there are more than 20% duplicate sequences in total this graph will be 
      flagged for a warning and for failure if it’s over 50%. There are many reasons why duplications can arise the most common examples
      are: enriched samples or by chance the same sequence was selected.
      <br> 
      for more information on this graph and how to interpret unexpected results please visit <a href="https://www.bioinformatics.babraham.ac.uk/projects/fastqc/Help/3%20Analysis%20Modules/8%20Duplicate%20Sequences.html" class="links_text_fastqc">Duplication Explanation</a>
      as it contains an extensive explanation on this graph.
   </p>
   <br>

   <h4 class="headers_text_fastqc"> Per base N content:</h4>
   <p class="text_fastqc">
      Sequencers can't always decide with confidence which of the bases it is and with substitutes
      it for N. It's normal for small amounts of N to appear because of quality loss near the end of the sequence. The 2 most common
      reasons for a large portion of N content are a bad quality or a biased sample in which case the sequencer can get confused and make 
      bad calls.
   </p>
   <br>

   <h4 class="headers_text_fastqc">Per base sequence quality:</h4>
   <p class="text_fastqc">
      This boxplot contains the quality per base.
      <br>
      Frequent problems
      <br>
      its commmon for reads to degrade in quality over the duration in long runs.
      There are a couple of different types of encoding that can be applied on the file, this can lead to false reports on
      this plot.
   </p>
   <br>

   <h4 class="headers_text_fastqc"> Per base sequence content:</h4>
   <p class="text_fastqc">
      In this graph you can see the proportions of the different nucleotide bases
      <br>
      In most cases there should be no bias between the complementing bases throughout the run, its worth mentioning that certain
      samples can produce biases in the beginning of a run based on the way they where prepared.
      <br>
      if the AT or the CG compensation is more then 10% appart it wil be flagged for warning and if its greater then 20% it will be flagged
      for failure.
      <br>
      Primers, adapters, and some samples based on their preparation can create biases.
   </p>
   <br>

   <h4 class="headers_text_fastqc"> Per sequence GC content:</h4>
   <p class="text_fastqc">
      In this plot you can see the CG content across the sequence
      <br>
      The peak of this graph should match the overall CG content of the genome you are testing. If this is not the case
      or there are multiple peaks the sample can be contaminated. The theoretical distribution is calculated based on the provided data.
      If this is weirdly shaped it can also indicate that the sample was contaminated.
      This plot will flag for warning if the total amount of deviations from the normal distribution exceeds 15% and if it exceeds 30% its
      flagged for failure.
   </p>
   <br>

   <h4 class="headers_text_fastqc"> Per sequence quality score:</h4>
   <p class="text_fastqc">This graph shows the overall quality distribution of the reads.<br>
      if the overall quality is low there might be systematic problem with the flow cell or a specific part of the run.
      As per babraham bioinformatics standards if the overall quality is below 27 the read is flagged for warning and if it’s below 20
      its flagged for failure
   </p>
   <br>
         
   <h4 class="headers_text_fastqc"> Sequence length distribution:</h4>
   <p class="text_fastqc">
      This plot shows the distribution of the lengt of the reads.
      <br>
      Warnings will appear when the sequence is not of the same length, and it will be failed if the length is 0.
      Warnings in this graph can sometimes be ignored if you expect different lengths in sequence, this could be because of the type of
      sequencing you have done or data trimming which can result into a difference in length.
   </p>
   <br>      

   <h4 class="source_text_fastqc"> Source:</h4>
   <p class="source_text_fastqc">If you require more detailed information please visit the official FASTQC website: 
      <a href="https://www.bioinformatics.babraham.ac.uk/projects/fastqc/Help/" class="links_text_fastqc">Babraham Bioinformatics</a>.
   </p>
   <br>
   <br>
{% endblock %}